presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -1484,4 +1493,59 @@ protocol::NodeStatus PrestoServer::fetchNodeStatus() {
   return nodeStatus;
 }
 
+void PrestoServer::convertToVeloxPlan(
+    proxygen::HTTPMessage* message,
+    proxygen::ResponseHandler* downstream,
+    const std::vector<std::unique_ptr<folly::IOBuf>>& body) {
+  protocol::ExecutionFailureInfo failure;
+  try {
+    auto headers = message->getHeaders();
+
+    std::string planFragmentJson = util::extractMessageBody(body);
+    protocol::PlanFragment planFragment = json::parse(planFragmentJson);
+
+    auto queryCtx = core::QueryCtx::create();
+    VeloxInteractiveQueryPlanConverter converter(queryCtx.get(), pool_.get());
+
+    // Create static taskId and empty TableWriteInfo needed for plan conversion
+    protocol::TaskId taskId = "velox-plan-conversion.0.0.0";
```

Comment:
I tried to make sense of why this is important to plan conversion and see only one place where its used for AssignUniqueId here : https://github.com/prestodb/presto/blob/master/presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp#L1425

It expects a certain format to convert to a PrestoTaskId
https://github.com/prestodb/presto/blob/master/presto-native-execution/presto_cpp/main/types/PrestoTaskId.h#L20

We could fake these values. But it is possible to come up with a reasonable PrestoTaskId for the overall query and stage that we are validating ? You would have to send it from the co-ordinator.

Reply:
My feeling is that given that we won't be able to create a "correct" task id (the scheduler is the source of truth for that), it's better to have one that's obviously fake.

Reply:
Just to add, I think we had discussed this some time ago - technically we should be able to do the plan conversion without requiring a task id too, but it's not worth altering that code when we could just create a dummy one.